domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputHasFractionalPartExactlyEqualToRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,246 @@
+  private val NON_NEGATIVE_VALUE_0 = createNonNegativeInt(value = 0)
+  private val WHOLE_NUMBER_123 = createWholeNumber(isNegative = false, value = 123)
+  private val WHOLE_NUMBER_321 = createWholeNumber(isNegative = false, value = 321)
+  private val FRACTION_2_OVER_4 = createFraction(isNegative = false, numerator = 2, denominator = 4)
+  private val FRACTION_1_OVER_2 = createFraction(isNegative = false, numerator = 1, denominator = 2)
+  private val MIXED_NUMBER_123_1_OVER_2 =
+    createMixedNumber(isNegative = false, wholeNumber = 123, numerator = 1, denominator = 2)
+  private val MIXED_NUMBER_123_1_OVER_3 =
+    createMixedNumber(isNegative = false, wholeNumber = 123, numerator = 1, denominator = 3)
+
+  @Inject
+  internal lateinit var fractionInputHasFractionalPartExactlyEqualToRuleClassifierProvider:
+    FractionInputHasFractionalPartExactlyEqualToRuleClassifierProvider
+
+  private val fractionalPartIsExactlyEqualClassifierProvider: RuleClassifier by lazy {
+    fractionInputHasFractionalPartExactlyEqualToRuleClassifierProvider.createRuleClassifier()
+  }
+
+  @Test
+  fun testEquals_wholeNumber123Answer_withWholeNumber123Input_bothValuesMatch() {
+    val inputs = mapOf("f" to WHOLE_NUMBER_123)
+
+    val matches =
+      fractionalPartIsExactlyEqualClassifierProvider.matches(
+        answer = WHOLE_NUMBER_123,
+        inputs = inputs
+      )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testEquals_wholeNumber321Answer_withWholeNumber123Input_bothValuesMatch() {
+    val inputs = mapOf("f" to WHOLE_NUMBER_123)
+
+    val matches =
+      fractionalPartIsExactlyEqualClassifierProvider.matches(
+        answer = WHOLE_NUMBER_321,
+        inputs = inputs
+      )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testEquals_fraction2Over4Answer_withFraction1Over2Input_bothValuesDoNotMatch() {
+    val inputs = mapOf("f" to FRACTION_1_OVER_2)
+
+    val matches =
+      fractionalPartIsExactlyEqualClassifierProvider.matches(
+        answer = FRACTION_2_OVER_4,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquals_mixedNum123And1Over2Answer_withMixedNum123And1Over3Input_bothValuesDoNotMatch() {
+    val inputs = mapOf("f" to MIXED_NUMBER_123_1_OVER_3)
+
+    val matches =
+      fractionalPartIsExactlyEqualClassifierProvider.matches(
+        answer = MIXED_NUMBER_123_1_OVER_2,
+        inputs = inputs
+      )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquals_mixedNum123And1Over2Answer_withFraction1Over2Input_bothValuesMatch() {
+    val inputs = mapOf("f" to FRACTION_1_OVER_2)
+
+    val matches =
+      fractionalPartIsExactlyEqualClassifierProvider.matches(
+        answer = MIXED_NUMBER_123_1_OVER_2,
+        inputs = inputs
+      )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testEquals_wholeNumberAnswer_withMixedNumberInput_bothValuesDoNotMatch() {
```

Comment:
Maybe also add a test for negatives (e.g. -1/2 and -123 1/2 don't have equal fractional parts, but -1/2 and -1/2, where the latter doubles as an identity check).